agent/src/softwarecontaineragent.cpp

```diff
@@ -119,94 +120,113 @@ bool SoftwareContainerAgent::triggerPreload()
     return true;
 }
 
-inline bool SoftwareContainerAgent::isIdValid(ContainerID containerID)
+void SoftwareContainerAgent::assertContainerExists(ContainerID containerID)
 {
-    return (containerID < INT32_MAX)
-        && (containerID >= 0)
-        && (1 == m_containers.count(containerID));
+    if (containerID >= INT32_MAX) {
+        std::string errorMessage("Invalid Container ID: "
+                                + std::to_string(containerID)
+                                + ". ID can not be greater than INT32");
+        log_error() << errorMessage;
+        throw SoftwareContainerError(errorMessage);
```

Comment:
I think the error message is going to be written to output at the place SoftwareContainerError is caught. So that can create duplicate error messages.  If this is true maybe it would be sensible to remover logs before throwing errors.